src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -179,6 +180,8 @@ std::unique_ptr<OptimizerContext> DawnCompiler::runOptimizer(std::shared_ptr<SIR
   optimizer->checkAndPushBack<PassStencilSplitter>(maxFields);
   optimizer->checkAndPushBack<PassTemporaryType>();
   optimizer->checkAndPushBack<PassTemporaryMerger>();
+  optimizer->checkAndPushBack<PassFieldVersioning>(
+      PassFieldVersioning::FieldVersioningPassMode::FM_FixAccess);
```

Comment:
To me, even if related to field versioning, the create version and fix access are different passes. They take different input of the IIR and they do different transformations. I propose to put the FixAccess mode as a separate pass (DoubleBufferCopyOfFieldVersions)

Reply:
let's briefly discuss in person